ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -873,47 +871,30 @@ getModSummaryRule :: Rules ()
 getModSummaryRule = do
     defineEarlyCutoff $ \GetModSummary f -> do
         session <- hscEnv <$> use_ GhcSession f
-        let dflags = hsc_dflags session
         (modTime, mFileContent) <- getFileContents f
         let fp = fromNormalizedFilePath f
         modS <- liftIO $ runExceptT $
                 getModSummaryFromImports session fp modTime (textToStringBuffer <$> mFileContent)
         case modS of
-            Right res@(ms,_) -> do
-                let fingerPrint = hash (computeFingerprint f (fromJust $ ms_hspp_buf ms) dflags ms, hashUTC modTime)
-                return ( Just (BS.pack $ show fingerPrint) , ([], Just res))
+            Right res -> do
+                bufFingerPrint <- liftIO $
+                    fingerprintFromStringBuffer $ fromJust $ ms_hspp_buf $ msrModSummary res
+                let fingerPrint = fingerprintFingerprints
+                        [ msrFingerprint res, bufFingerPrint ]
+                return ( Just (fingerprintToBS fingerPrint) , ([], Just res))
             Left diags -> return (Nothing, (diags, Nothing))
 
     defineEarlyCutoff $ \GetModSummaryWithoutTimestamps f -> do
         ms <- use GetModSummary f
         case ms of
-            Just res@(msWithTimestamps,_) -> do
-                let ms = msWithTimestamps {
+            Just res@ModSummaryResult{..} -> do
+                let ms = msrModSummary {
                     ms_hs_date = error "use GetModSummary instead of GetModSummaryWithoutTimestamps",
                     ms_hspp_buf = error "use GetModSummary instead of GetModSummaryWithoutTimestamps"
                     }
-                dflags <- hsc_dflags . hscEnv <$> use_ GhcSession f
-                let fp = BS.pack $ show $ hash (computeFingerprint f (fromJust $ ms_hspp_buf msWithTimestamps) dflags ms)
-                return (Just fp, ([], Just res))
+                    fp = fingerprintToBS msrFingerprint
```

Comment:
Do we not include the buffer in the fingerprint anymore?

Reply:
This was a bug. 
The buffer is included in the `GetModSummary` fingerprint, but should have never been included in the `GetModSummaryWithoutTimestamps` fingerprint, which doesn't include the buffer.

Reply:
The deleted line is:

```
                 let fp = BS.pack $ show $ hash (computeFingerprint f (fromJust $ ms_hspp_buf msWithTimestamps) dflags ms)
```

Seems like the buffer was included in the fingerprint?

Reply:
That was a bug, because in line 893 the buffer was elided